packages/app-degree-pages/src/core/utils/fetch-plan-codes.js

```diff
@@ -0,0 +1,23 @@
+// @ts-check
+
+/**
+ * Fetches details for a list of academic plan codes.
+ * @param {string[]} acadPlanCodes Array of academic plan codes.
+ * @returns {Promise<Object[]>} A promise that resolves to an array of academic plan details.
+ */
+async function fetchAcademicPlans(acadPlanCodes) {
```

Comment:
Mentioned before, how are we handling errors? timeouts?

Reply:
5s, but maybe 10s would lead to less likelihood of false positives while still having a safeguard.